File: app/src/main/java/com/bq/zowi/models/commands/ToneCommand.java
package com.bq.zowi.models.commands;

import com.bq.zowi.models.commands.Command;

/* JADX INFO: loaded from: classes.dex */
public class ToneCommand extends MotionlessCommand {
    private static final String TONE = "T";
    private static final Command.Action[] TONE_ACTIONS = {Command.Action.TONE};
    private int frecuency;

    public ToneCommand(int frecuency, long duration) {
        setAction(Command.Action.TONE);
        this.frecuency = frecuency;
        this.duration = Long.valueOf(duration);
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command.Action[] getAllowedActions() {
        return TONE_ACTIONS;
    }

    @Override // com.bq.zowi.models.commands.Command
    public String getCommandValue() {
        return "T " + this.frecuency + " " + this.duration + Command.CRLN;
    }

    @Override // com.bq.zowi.models.commands.Command
    public Command copy() {
        return new ToneCommand(this.frecuency, this.duration.longValue());
    }
}

File: zowi-core/src/main/kotlin/com/bq/zowi/models/commands/ToneCommand.kt
package com.bq.zowi.models.commands

class ToneCommand(frequency: Int, duration: Long) : MotionlessCommand() {

    private val frequency: Int = frequency

    init {
        setAction(Command.Action.TONE)
        setDurationRaw(duration)
    }

    override fun getAllowedActions(): Array<Command.Action> = TONE_ACTIONS
    override fun getCommandValue(): String = "T ${frequency} ${_duration}${Command.CRLN}"
    override fun copy(): Command = ToneCommand(frequency, _duration!!)

    companion object {
        private val TONE_ACTIONS = arrayOf(Command.Action.TONE)
    }
}
